tests/installed/libinsttest.sh

```diff
@@ -20,6 +20,49 @@
 dn=$(dirname $0)
 . ${dn}/libtest-core.sh
 
+# Copy of bits from tap-test
+test_tmpdir=
+function _tmpdir_cleanup () {
+    if test -n "${test_tmpdir}" && test -f ${test_tmpdir}/.testtmp; then
+	      rm "${test_tmpdir}" -rf
+    fi
+}
+prepare_tmpdir() {
+    test_tmpdir=$(mktemp -d)
+    touch ${test_tmpdir}/.testtmp
+    cd ${test_tmpdir}
+    echo ${test_tmpdir}
+}
+
+# This is copied from flatpak/flatpak/tests/test-webserver.sh
+run_tmp_webserver() {
+    dir=$1
+
+    test -n ${test_tmpdir}
+
+    cd ${dir}
+    env PYTHONUNBUFFERED=1 setsid python -m SimpleHTTPServer 0 &>${test_tmpdir}/httpd-output &
+    cd -
+    child_pid=$!
+
+    for x in $(seq 50); do
+        # Snapshot the output
+        cp ${test_tmpdir}/httpd-output{,.tmp}
+        # If it's non-empty, see whether it matches our regexp
+        if test -s ${test_tmpdir}/httpd-output.tmp; then
+            sed -e 's,Serving HTTP on 0.0.0.0 port \([0-9]*\) \.\.\.,\1,' < ${test_tmpdir}/httpd-output.tmp > ${test_tmpdir}/httpd-port
+            if ! cmp ${test_tmpdir}/httpd-output.tmp ${test_tmpdir}/httpd-port 1>/dev/null; then
+                # If so, we've successfully extracted the port
+                break
+            fi
+        fi
+        sleep 0.1
+    done
+    port=$(cat ${test_tmpdir}/httpd-port)
+    echo "http://127.0.0.1:${port}" > ${test_tmpdir}/httpd-address
+    echo "$child_pid" > ${test_tmpdir}/httpd-pid
```

Comment:
Yeah, it's not torn down.  Broadly speaking installed tests that run as root should probably be assumed to be destructive, and hence we don't do VM reuse.  But as a convenience we do try to support re-running the tests to help debug in the local scenario.

Basically I did make a change to `rm repo -rf` at the start for this, but I'm not sure it's worth trying to invest more in this versus just having the local dev path use a VM-snapshotting framework like vagrant.